src/exportJarFileCommand.ts

```diff
@@ -0,0 +1,337 @@
+                const jarQuickPickItem: IJarQuickPickItem = {
+                    label: getName(mainMethod),
+                    description: mainMethod.name,
+                };
+                pickNodes.push(jarQuickPickItem);
+            }
+        }
+        if (pickNodes.length === 0) {
+            return resolve("");
+        } else {
+            const noMainClassItem: IJarQuickPickItem = {
+                label: "No main class",
+                description: "",
+            };
+            pickNodes.push(noMainClassItem);
+            const pickBox = createPickBox("Export Jar - Determine main class", "Select the main class...", pickNodes, false, pickSteps.length);
+            pickBox.onDidTriggerButton((item) => {
+                if (item === QuickInputButtons.Back) {
+                    reject(InputFlowAction.back);
+                    pickBox.dispose();
+                }
+            });
+            pickBox.onDidAccept(() => {
+                pickSteps.push(ExportSteps.ResolveMainMethod);
+                resolve(pickBox.selectedItems[0].description);
+                pickBox.dispose();
+            });
+            pickBox.onDidHide(() => {
+                reject();
+                pickBox.dispose();
+            });
+            pickBox.show();
+        }
+    });
+}
+
+function failMessage(message: string) {
+    window.showErrorMessage(message, "Done");
+}
+
+function successMessage(outputFileName: string) {
+    let openInExplorer: string;
+    if (platform() === "win32") {
+        openInExplorer = "Reveal in File Explorer";
+    } else if (platform() === "darwin") {
+        openInExplorer = "Reveal in Finder";
+    } else {
+        openInExplorer = "Open Containing Folder";
+    }
+    window.showInformationMessage("Successfully exported jar to" + EOL + outputFileName,
+        openInExplorer, "Done").then((messageResult) => {
+            if (messageResult === openInExplorer) {
+                commands.executeCommand("revealFileInOS", Uri.file(outputFileName));
+            }
+        });
+}
+
+async function generateOutClassPath(pickSteps: string[], rootNodes: INodeData[], projectPath: string): Promise<string[] | undefined> {
+    return new Promise<string[] | undefined>(async (resolve, reject) => {
+        const extension: Extension<any> | undefined = extensions.getExtension("redhat.java");
+        const extensionApi: any = await extension?.activate();
+        const outClassPaths: string[] = [];
+        const setUris: Set<string> = new Set<string>();
+        let pickDependencies: IJarQuickPickItem[] = [];
+        const pickedDependencies: IJarQuickPickItem[] = [];
+        for (const rootNode of rootNodes) {
+            const classPaths: ClasspathResult = await extensionApi.getClasspaths(rootNode.uri, { scope: "runtime" });
+            pickDependencies = pickDependencies.concat(generateDependencies(classPaths.classpaths, setUris, projectPath, true));
```

Comment:
I think `push` would be better: https://dev.to/uilicious/javascript-array-push-is-945x-faster-than-array-concat-1oki